docs/migration-guide.md

```diff
@@ -19,6 +19,7 @@ The type `sds` is removed from the public API.
 
 * `redisFormatSdsCommandArgv` removed from API. Can be replaced with `valkeyFormatCommandArgv`.
 * `redisFreeSdsCommand` removed since the `sds` type is for internal use only.
+* `redisAsyncSetConnectCallbackNC` removed and `valkeyAsyncSetConnectCallback` accepts the non-const callback function prototype.
```

Comment:
Shouldn't we also mention `redisAsyncSetConnectCallback` on it's own line?

It's renamed in common way to `valkeyAsyncSetConnectCallback`, but the const modifier is removed.